src/components/project/NewProject/NewProject.tsx

```diff
@@ -127,87 +127,66 @@ const NewProject: FC<Props> = ({
-        file: null,
-        defaultFiles: [
-          {
-            id: '',
-            parent: null,
-            path: defaultFileName,
-            type: 'file' as const,
-            name: defaultFileName,
-            content: decodeBase64(code),
-          },
-        ],
-        isTemporary: true,
-      });
-      const finalQueryParam = router.query;
-      delete finalQueryParam.code;
-      delete finalQueryParam.lang;
-      router.replace({ query: finalQueryParam }).catch(() => {});
-      openTab(defaultFileName, `${baseProjectPath}/temp/${defaultFileName}`);
-    } catch (error) {
-      if (error instanceof Error) {
-        createLog(error.message, 'error');
-        return;
+  const onRouterReady = async () => {
+    if (codeToImport) {
+      // Default to 'func' as the language if none is provided in the query parameters.
+      // This ensures backward compatibility for cases where the language was not included in the query params initially.
+      try {
+        await importEncodedCode(codeToImport, importLanguage ?? 'func');
+      } catch (error) {
+        if (error instanceof Error) {
+          createLog(error.message, 'error');
+          return;
+        }
+        createLog('Error in importing code', 'error');
       }
+      return;
     }
-  };
 
-  useEffect(() => {
-    if (codeToImport) {
-      importFromCode(codeToImport as string);
+    // When the IDE starts without any projects and no import URL is provided,
+    // or if projects already exist but the import URL is missing or the modal is inactive,
+    // skip setting up the New Project dialog for import.
+    if (
+      (projects.length == 0 && !importURL && !active) ||
+      (projects.length !== 0 && (!importURL || !active))
+    ) {
       return;
     }
 
-    if (!importURL || !active) {
-      return;
+    // If there are no projects but an import URL is provided, set the project type to git
+    if (projects.length === 0 && importURL) {
+      setNewProjectType('git');
     }
 
     form.setFieldsValue({
       template: 'import',
-      githubUrl: importURL || '',
+      githubUrl: importURL ?? '',
       name: projectName ?? '',
       language: importLanguage ?? 'func',
     });
     setIsActive(true);
-    const finalQueryParam = router.query;
-    delete finalQueryParam.importURL;
-    delete finalQueryParam.name;
-    router.replace({ query: finalQueryParam }).catch(() => {});
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [importURL, projectName, form, codeToImport]);
 
-  const closeModal = () => {
-    setIsActive(false);
+    // Wait for the form to be set up before removing the import query parameters
+    await delay(100);
+    removeImportParams();
   };
 
   useEffect(() => {
-    EventEmitter.on('ONBOARDING_NEW_PROJECT', () => {
-      setIsActive(true);
-    });
-    return () => {
-      EventEmitter.off('ONBOARDING_NEW_PROJECT');
-    };
-  }, []);
+    if (!router.isReady) return;
+    onRouterReady();
```

Comment:
This depends on `onRouterReady`, but doesn't have it in the list of dependencies. The `onRouterReady` depends on `codeToImport`. It's unclear whether `useEffect` is supposed to be run whenever `codeToImport` changes or not.

Please, use `useCallback` for event handlers defined inside of a component and [exhaustive deps](https://stackoverflow.com/questions/58866796/understanding-the-react-hooks-exhaustive-deps-lint-rule), that's how hooks are designed to work by React team.

How it could work: `onRouterReady` is `useCallback` that depends on `codeToImport`, `useEffect` depends on `onRouterReady`. If `codeToImport` is non-empty, we remove it from URL and start processing it. Removal of `codeToImport` triggers the update chain again, but this time it's empty, and we don't proceed to import.